lib/wasm/WebAssemblyJavascriptGenerator.js

```diff
@@ -8,59 +8,96 @@ const Generator = require("../Generator");
 const Template = require("../Template");
 const { RawSource } = require("webpack-sources");
 const WebAssemblyImportDependency = require("../dependencies/WebAssemblyImportDependency");
-
-function generateInitParams(module) {
-	const list = [];
-
-	for (const dep of module.dependencies) {
-		if (dep instanceof WebAssemblyImportDependency) {
-			if (dep.description.type === "GlobalType") {
-				const exportName = dep.name;
-				const usedName = dep.module && dep.module.isUsed(exportName);
-
-				if (dep.module === null) {
-					// Dependency was not found, an error will be thrown later
-					continue;
-				}
-
-				if (usedName !== false) {
-					list.push(
-						`__webpack_require__(${JSON.stringify(
-							dep.module.id
-						)})[${JSON.stringify(usedName)}]`
-					);
-				}
-			}
-		}
-	}
-
-	return list;
-}
+const WebAssemblyExportImportedDependency = require("../dependencies/WebAssemblyExportImportedDependency");
 
 class WebAssemblyJavascriptGenerator extends Generator {
 	generate(module, dependencyTemplates, runtimeTemplate) {
 		const initIdentifer = Array.isArray(module.usedExports)
 			? Template.numberToIdentifer(module.usedExports.length)
 			: "__webpack_init__";
 
-		const generateImports = () => {
-			const modules = new Map();
-			for (const dep of module.dependencies) {
-				if (dep.module) modules.set(dep.module, dep.userRequest);
-			}
-			return Template.asString(
-				Array.from(modules, ([m, r]) => {
-					return `${runtimeTemplate.moduleRaw({
-						module: m,
-						request: r
-					})};`;
-				})
-			);
-		};
+		let needExportsCopy = false;
```

Comment:
It's not really a copy? It's more a rewire or a connection

Reply:
This variable is about a shallow copy of the exports object.

If false it can use `module.exports = instance.exports`.
If true it has to use `for(var key in instance.exports) exports[key] = instance.exports[key]`

Reply:
ah yes, it's https://github.com/webpack/webpack/pull/7467/files#diff-2812aaa32808448f93ae4969dc9df373R115 thanks